packages/datatrak-web-server/src/routes/SubmitSurvey/buildUpsertEntity.ts

```diff
@@ -4,37 +4,69 @@
  */
 
 import { generateId } from '@tupaia/database';
-import { DatatrakWebSubmitSurveyRequest, Entity, SurveyScreenComponentConfig } from '@tupaia/types';
+import { ajvValidate, objectEntries } from '@tupaia/tsutils';
+import {
+  DatatrakWebSubmitSurveyRequest,
+  Entity,
+  SurveyScreenComponentConfig,
+  EntityUpdate,
+  EntityUpdateSchema,
+  EntityQuestionConfigFieldValue,
+} from '@tupaia/types';
 import { DatatrakWebServerModelRegistry } from '../../types';
 
 type Answers = DatatrakWebSubmitSurveyRequest.ReqBody['answers'];
 
+const isQuestionValue = (
+  configValue?: EntityQuestionConfigFieldValue,
+): configValue is { questionId: string } => {
+  return !!(configValue && typeof configValue === 'object' && 'questionId' in configValue);
+};
+
 export const buildUpsertEntity = async (
   models: DatatrakWebServerModelRegistry,
   config: SurveyScreenComponentConfig,
   questionId: string,
   answers: Answers,
   countryId: Entity['id'],
 ) => {
-  const entityId = (answers[questionId] || generateId()) as Entity['id'];
-  const entity = { id: entityId } as Entity;
-  const fields = config?.entity?.fields || {};
-
-  for (const [fieldName, value] of Object.entries(fields)) {
-    // Value is not defined, skip
-    if (value === undefined) {
-      return;
-    }
+  const entityId = answers[questionId] || generateId();
```

Comment:
Might be a good idea for us to do some regression testing of creating entities via datatrak as part of this card?